### PR TITLE
Shell, PHP, Python,JS snippets

### PR DESCRIPTION
**Pull request by KBbitsP**
Shell, PHP, Python,JS snippets

### CODE REVIEW

File: source/includes/_security.md [hunk 1]
```diff
@@ -8,18 +8,405 @@ Please refer to our [RBAC manual](https://docs.killbill.io/latest/user_managemen
...
+    "http://localhost:8080/1.0/kb/security/permissions"
+```
+
+```java
+import org.killbill.billing.client.api.gen.SecurityApi;
+import org.killbill.billing.client.model.Strings;
+
+SecurityApi securityApi = new SecurityApi(killBillHttpClient);
+
+Strings currentUserPermissions = securityApi.getCurrentUserPermissions(requestOptions);
```
Comment (reshmabidikar) [inline source/includes/_security.md:26]: Typo here: should be `String currentUserPermissions`

File: source/includes/_security.md [hunk 2]
```diff
@@ -8,18 +8,405 @@ Please refer to our [RBAC manual](https://docs.killbill.io/latest/user_managemen
...
+```shell
+curl -v \
+     -u admin:password \
+     -H "X-Killbill-ApiKey: bob" \
+     -H "X-Killbill-ApiSecret: lazar" \
+     -H "Accept: application/json" \
+    "http://localhost:8080/1.0/kb/security/subject"
+```
+
+```java
```
Comment (reshmabidikar) [inline source/includes/_security.md:85]: missing java code

File: source/includes/_security.md [hunk 3]
```diff
@@ -8,18 +8,405 @@ Please refer to our [RBAC manual](https://docs.killbill.io/latest/user_managemen
...
+```
+
+```ruby
+```
```
Comment (reshmabidikar) [inline source/includes/_security.md:89]: missing ruby code

File: source/includes/_security.md [hunk 4]
```diff
@@ -8,18 +8,405 @@ Please refer to our [RBAC manual](https://docs.killbill.io/latest/user_managemen
...
+$body->setUsername('TestUserName');
+$body->setPassword('TestUserPassword');
+$body->setRoles(['TestRole']);
+
+$apiInstance->addUserRoles($body,$xKillbillCreatedBy,$xKillbillReason,$xKillbillComment);
+```
+> Example Response:
+
+```json
+
```
Comment (reshmabidikar) [inline source/includes/_security.md:184]: missing response

File: source/includes/_security.md [hunk 5]
```diff
@@ -102,8 +505,38 @@ curl -v \
 ```
```
Comment (reshmabidikar) [inline source/includes/_security.md:506]: This looks like a new endpoint, heading is missing here.

File: source/includes/_security.md [hunk 6]
```diff
@@ -144,12 +577,39 @@ curl -v \
...
```
Comment (reshmabidikar) [inline source/includes/_security.md:578]: Heading missing here

**Review by reshmabidikar — COMMENTED** [review]
Since this is a big PR with a lot of changes, it's difficult to review this properly. I've added some comments for issues that I noticed. Also, some overall comments:
1. Java/Ruby code is missing for some endpoints
2. Endpoint descriptions/HTTP Request/Request Body/Query Parameters/Returns sections are missing for all endpoints
3. Endpoint headers are missing

Also, I've created a new branch called `security-resource-update`. Would request you to point your PR to this branch. Let's merge into `master` only when the changes are in a good state.

For large tasks such as this, would request you to please create smaller PRs addressing a defined subset of the overall task.

File: source/includes/_security.md (hunk 1 above)
Comment (KBbitsP) [reply source/includes/_security.md:26]: > Typo here: should be `String currentUserPermissions`

Hi @reshmabidikar this looks like the correct one

image

**Review by reshmabidikar — APPROVED** [review]
Approving for now. Let's review again once all the changes are completed and merged.

**Pull request merged**